sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2018,6 +2018,62 @@ class Analyzer(
         throw new AnalysisException("Only one generator allowed per select clause but found " +
           generators.size + ": " + generators.map(toPrettySQL).mkString(", "))
 
+      case Aggregate(_, aggList, _) if aggList.exists(hasNestedGenerator) =>
+        val nestedGenerator = aggList.find(hasNestedGenerator).get
+        throw new AnalysisException("Generators are not supported when it's nested in " +
+          "expressions, but got: " + toPrettySQL(trimAlias(nestedGenerator)))
+
+      case Aggregate(_, aggList, _) if aggList.count(hasGenerator) > 1 =>
+        val generators = aggList.filter(hasGenerator).map(trimAlias)
+        throw new AnalysisException("Only one generator allowed per aggregate clause but found " +
+          generators.size + ": " + generators.map(toPrettySQL).mkString(", "))
+
+      case agg @ Aggregate(groupList, aggList, child) if aggList.forall {
+          case AliasedGenerator(generator, _, _) => generator.childrenResolved
```

Comment:
please take a look at the object `AliasedGenerator`. There is no `AliasedGenerator` instance. It's just an `unapply` method. And we can use `hasGenerator` here.

Reply:
Ah you are right, `Generator` instance is resolved, but the structure `Alias(GeneratorOuter(generator)`, the `Alias` node and `GeneratorOuter` node is unresolved.
So we still cannot use the judgment `if aggList.forall(_.resolved)`.